massa-protocol-worker/src/node_info.rs

```diff
@@ -133,9 +135,9 @@ impl NodeInfo {
     pub fn insert_known_ops(&mut self, ops: Set<OperationId>, max_ops_nb: usize) {
         for operation_id in ops.into_iter() {
             if self.known_operations.insert(operation_id) {
-                self.known_operations_queue.push_front(operation_id);
-                if self.known_operations_queue.len() > max_ops_nb {
-                    if let Some(op_id) = self.known_operations_queue.pop_back() {
+                self.known_operations_queue.push_back(operation_id);
+                while self.known_operations_queue.len() > max_ops_nb {
```

Comment:
Why the `while` loop? You could just just a simple `if self.known_operations_queue.len() > max_ops_nb`.

Reply:
indeed no loop should be required, and this doesn't add any overhead. I just sends the following message to the developer: "after this loop you know that there is no excess element anymore". The behavior is exactly the same.

Reply:
I don't see how this sends a clearer message to the developer then a simple if, and I find it actually confusing since a loop seems to imply some kind of iteration(of more than one element), which never actually takes place. 

The reason why a loop is used for example when using a condvar, is to allow for spurious wake-ups, which is not an issue here.